modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -213,10 +214,11 @@ const deleteSelectedContent = (editor: Editor) => {
   }
 };
 
-export const insertHtmlAtCaret = (editor: Editor, value: string, details) => {
-  let parentNode, rootNode, args;
-  let marker, rng, node;
-  const selection: EditorSelection = editor.selection, dom = editor.dom;
+export const insertHtmlAtCaret = (editor: Editor, value: string, details: InsertContentDetails): void => {
+  let parentNode;
+  let rng, node;
```

Comment:
Wondering if these can be given types or do they need to be `any`?

Reply:
The types for these are complicated, partly because there's still IE 8 code in here. As I said in the description, it's known that some of these won't have types as the intention at this stage is to only fix the APIs not the internals.